TelCo.ColorCoder/ClassforGetPairNumberFromColor.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+	public class ClassforGetPairNumberFromColor
+	{
+        internal static int GetPairNumberFromColor(ColorPair pair)
+        {
+            // Find the major color in the array and get the index
+            int majorIndex = -1;
+            for (int i = 0; i < Program.getcolorMapMajorlen(); i++)
+            {
+                if (Program.getcolorMapMajor(i) == pair.majorColor)
+                {
+                    majorIndex = i;
+                    break;
+                }
+            }
+
+            // Find the minor color in the array and get the index
+            int minorIndex = -1;
+            for (int i = 0; i < Program.getcolorMapMinorlen(); i++)
```

Comment:
This `for` loop is nearly a duplicate of the previous one. Try removing the duplication

Reply:
done